projects/igniteui-angular/src/lib/grids/columns/column.component.ts

```diff
@@ -405,23 +405,67 @@ export class IgxColumnComponent implements AfterContentInit, OnDestroy, ColumnTy
     }
 
     /**
-     * @hidden
+     * Emitted when the column is hidden or shown.
+     *
+     * ```html
+     * <igx-column (hiddenChange)="hiddenChange($event)">
+     * </igx-column>
+     * ```
+     *
+     * @memberof IgxColumnComponent
      */
     @Output()
     public hiddenChange = new EventEmitter<boolean>();
 
-    /** @hidden */
+    /**
+     * Emitted when the column expanded or collapsed.
+     *
+     * ```html
+     * <igx-column (expandedChange)="expandedChange($event)">
+     * </igx-column>
+     * ```
+     *
+     * @memberof IgxColumnComponent
+     */
     @Output()
     public expandedChange = new EventEmitter<boolean>();
 
-    /** @hidden */
+    /**
+     * Emitted when the column property `collapsible` changes.
+     *
+     * ```html
+     * <igx-column (collapsibleChange)="collapsibleChange($event)">
+     * </igx-column>
+     * ```
+     *
+     * @memberof IgxColumnComponent
+     */
     @Output()
     public collapsibleChange = new EventEmitter<boolean>();
-    /** @hidden */
+
+    /**
+     * Emitted when the column property `visibleWhenCollapsed` changes.
+     *
+     * ```html
+     * <igx-column (visibleWhenCollapsedChange)="visibleWhenCollapsedChange($event)">
+     * </igx-column>
+     * ```
+     *
+     * @memberof IgxColumnComponent
+     */
     @Output()
     public visibleWhenCollapsedChange = new EventEmitter<boolean>();
```

Comment:
Okay so far the others make sense - `hiddenChange`, `expandedChange` - both meaningful _user_ interactions that can modify the state.
Might be missing something but don't see `collapsible` or `visibleWhenCollapsed` being modified by an interaction. Those sound like pure configuration to me and as such I'm not even sure why those events exist in the first place and at the very least don't think they need to be public.

Reply:
I'll revert it for `collapsible` and `visibleWhenCollapsed`